feat: Use float instead of int for GCE_METADATA_TIMEOUT

ludoch opened the pull request:
Move the logic to read GCE_METADATA_TIMEOUT env variable so we can better control it in app_identity component.

Related internal change is 607513254.

ludoch committed:
Update _metadata.py

Move the logic to read GCE_METADATA_TIMEOUT env variable so we can better control it in app_identity component.

Related internal change is 607513254.

estrellis commented:
The change LGTM

clundin25 commented on google/auth/compute_engine/_metadata.py:
```diff
@@ -119,6 +116,11 @@ def ping(request, timeout=_METADATA_DEFAULT_TIMEOUT, retry_count=3):
     #       "unlikely".
+    try:
+        timeout = float(os.getenv("GCE_METADATA_TIMEOUT", timeout))
+    except ValueError:  # pragma: NO COVER
+        pass
```
I would not capture a ValueError here. Additionally, the timeout variable would not be usable, so the downstream call would have problms.

estrellis replied:
FWIW the original code caught the ValueError. The value is set by the user.
I am thinking if the Exception happens timeout will retain the passed in/default value

clundin25 replied:
In that case, can you remove the pragma, and include test coverage for this change?

ludoch committed:
Update _metadata.py

clundin25 commented on google/auth/compute_engine/_metadata.py:
```diff
@@ -53,10 +53,7 @@
-    _METADATA_DEFAULT_TIMEOUT = int(os.getenv("GCE_METADATA_TIMEOUT", 3))
-except ValueError:  # pragma: NO COVER
-    _METADATA_DEFAULT_TIMEOUT = 3
+_METADATA_DEFAULT_TIMEOUT = 3
```
Can you revert this change? It seems safe to keep as is.

clundin25 commented on google/auth/compute_engine/_metadata.py:
```diff
@@ -119,6 +116,9 @@ def ping(request, timeout=_METADATA_DEFAULT_TIMEOUT, retry_count=3):
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
+    if timeout is None:
+        timeout = float(os.getenv("GCE_METADATA_TIMEOUT", _METADATA_DEFAULT_TIMEOUT))
```
Please add a test case, and then this LGTM.

ludoch committed:
Merge branch 'googleapis:main' into main

ludoch committed:
Merge branch 'googleapis:main' into main

ludoch committed:
Change timeout behavior and add a test for metadata server access. (#1)

* Add a test with new _metadata.py changes

* Add tests for  metadata changes.

ludoch committed:
add env var (#2)

* chore(python): update dependencies in /.kokoro (#1499)

Source-Link: https://github.com/googleapis/synthtool/commit/db94845da69ccdfefd7ce55c84e6cfa74829747e
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-python:latest@sha256:a8a80fc6456e433df53fc2a0d72ca0345db0ddefb409f1b75b118dfd1babd952

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Anthonios Partheniou <partheniou@google.com>

* Add a test with new _metadata.py changes

* Add tests for  metadata changes.

* chore(python): bump idna from 3.4 to 3.7 in .kokoro (#1517)

Source-Link: https://github.com/googleapis/synthtool/commit/d50980e704793a2d3310bfb3664f3a82f24b5796
Post-Processor: gcr.io/cloud-devrel-public-resources/owlbot-python:latest@sha256:5a4c19d17e597b92d786e569be101e636c9c2817731f80a5adec56b2aa8fe070

Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Anthonios Partheniou <partheniou@google.com>

* Update environment_vars.py

---------

Co-authored-by: gcf-owl-bot[bot] <78513119+gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>
Co-authored-by: Anthonios Partheniou <partheniou@google.com>

chalmerlowe committed:
removes several files that no longer exist in main

chalmerlowe committed:
removes additional files that are not longer in the main branch

chalmerlowe committed:
Merge branch 'main' into main

chalmerlowe committed:
Merge branch 'main' into main

chalmerlowe committed:
Merge branch 'main' into main
